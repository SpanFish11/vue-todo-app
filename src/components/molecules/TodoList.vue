<script setup>
import { computed, ref } from 'vue'
import TodoItem from '@/components/molecules/TodoItem.vue'
import draggable from 'vuedraggable'
import { useStore } from 'vuex'
import EmptyTodo from '@/components/atoms/EmptyTodo.vue'

const drag = ref(false)

const store = useStore()
const todos = computed(() => store.getters.getTodos)
</script>

<template>
  <draggable
    v-if="todos.length"
    :list="todos"
    @start="drag = true"
    @end="drag = false"
    item-key="id"
    tag="ul"
    class="list-group list-group-flush"
  >
    <template #item="{ element }">
      <TodoItem :todo="element" />
    </template>
  </draggable>

  <EmptyTodo v-else />
</template>
