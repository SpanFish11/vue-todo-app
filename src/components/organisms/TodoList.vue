<script setup>
import { computed, ref } from 'vue'
import TodoItem from '@/components/molecules/TodoItem.vue'
import draggable from 'vuedraggable'
import { useStore } from 'vuex'

const drag = ref(false)

const store = useStore()
const todos = computed(() => store.getters.getTodos)
</script>

<template>
  <draggable :list="todos" @start="drag = true" @end="drag = false" item-key="id" class="list">
    <template #item="{ element }">
      <TodoItem :todo="element" />
    </template>
  </draggable>
</template>

<style scoped lang="scss">
.list {
  display: flex;
  width: 650px;
  flex-direction: column;
  border-radius: 0.4em;
  background-color: #ffffff;

  > :first-child {
    border-top-left-radius: 0.4em;
    border-top-right-radius: 0.4em;
  }

  > :last-child {
    border-bottom-left-radius: 0.4em;
    border-bottom-right-radius: 0.4em;
    border-bottom: none;
  }
}
</style>
