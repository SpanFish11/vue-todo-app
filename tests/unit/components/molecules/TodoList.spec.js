import { describe, expect, it } from '@jest/globals'
import { mount } from '@vue/test-utils'
import { createStore } from 'vuex'
import TodoList from '@/components/molecules/TodoList.vue'

describe('Given the TodoList component', () => {
  it('should render', () => {
    const store = createStore({})
    store.getters.getTodos = [
      { title: 'title-1', completed: false, id: '1' },
      { title: 'title-2', completed: false, id: '2' },
      { title: 'title-3', completed: true, id: '3' },
    ]

    const wrapper = mount(TodoList, {
      global: {
        plugins: [store],
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
