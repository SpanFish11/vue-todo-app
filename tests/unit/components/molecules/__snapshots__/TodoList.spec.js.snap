// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the TodoList component should render 1`] = `
<ul
  class="list-group list-group-flush"
>
  <div
    class="list-group-item d-flex align-items-center"
    data-draggable="true"
  >
    <input
      class="form-check-input ms-2"
      type="checkbox"
    />
    <span
      class="ms-4 custom-span"
    >
      title-1
    </span>
    <button
      class="button item__button--delete btn-close ms-auto"
    />
  </div>
  <div
    class="list-group-item d-flex align-items-center"
    data-draggable="true"
  >
    <input
      class="form-check-input ms-2"
      type="checkbox"
    />
    <span
      class="ms-4 custom-span"
    >
      title-2
    </span>
    <button
      class="button item__button--delete btn-close ms-auto"
    />
  </div>
  <div
    class="list-group-item d-flex align-items-center"
    data-draggable="true"
  >
    <input
      checked=""
      class="form-check-input ms-2"
      type="checkbox"
    />
    <span
      class="todo-item-completed ms-4 custom-span"
    >
      title-3
    </span>
    <button
      class="button item__button--delete btn-close ms-auto"
    />
  </div>
</ul>
`;
